src/main/java/org/candlepin/service/impl/DefaultUserServiceAdapter.java

```diff
@@ -214,10 +214,9 @@ public List<? extends UserInfo> listUsers() {
     @Override
     public boolean validateUser(String username, String password) {
         User user = this.userCurator.findByLogin(username);
-        String hashedPassword = Util.hash(password);
 
-        if (user != null && password != null && hashedPassword != null) {
-            return hashedPassword.equals(user.getHashedPassword());
+        if (user != null && password != null) {
+            return OpenBSDBCrypt.checkPassword(user.getPassword(), password.toCharArray());
```

Comment:
Where is this check being used today? Have we verified what happens with existing hashes? Does the registration method support generating passwords with this format so the passwords will be properly checked?

Reply:
Note that this doesn't need to be backwards compatible, because candlepin doesn't manage passwords/users in a production setup (both Satellite and Hosted use their own user/passwd management).
We only  generate passwords for dev/test purposes

Reply:
Where this is used today, is when BasicAuth is used, and the DefaultUserServiceAdapter.java is only used in a non-prod setup

Reply:
This is only used for dev/test purposes. This is not being used on prod. We are only doing this to satisfy the static analyzer. There should be no existing hashes stored in db(They are stored just when tests runs). So it should not happened that we would be comparing SHA-I and BCrypt

Reply:
It's not strictly necessary to change stuff just because a tool said to do so. If we don't actually use this anywhere, then maybe a better answer is to either flag it as "won't fix" in the tool of choice, or completely isolate it so it doesn't ship and *can't* be used in production; thus eliminating the concern entirely.